python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -360,6 +358,28 @@ class RandomForestClassifier(Base):
         else:
             free(<RandomForestMetaData[double, int]*><size_t> self.rf_forest64)
 
+    def _new_forest_data(self):
+        # Reset the data
+        # Cannot include cdef in if block (compile time error)
+        cdef RandomForestMetaData[float, int] *rf_forest = \
```

Comment:
to merge these two functions (which seem like one function conceptually) maybe get rid of the cdef? Can you just do: ```
 self.rf_forest = new RandomForestMetaData[float, int]()
```

Reply:
That resulted in an error, but I updated it to check the case for
```
if self.n_cols==None:
    return
```
To add the cdef statements in the same function